phptravels website/test_hotel_subpage.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+import time
+from selenium.webdriver.common.action_chains import ActionChains
+from selenium.webdriver.common.keys import Keys
+
+# 1. Log in
+# 2. Navigate to "Hotels" subpage
+# 3. Test different boundary values in the search element
+
+
+def room_fillout(driver, get_xpaths, options):
+    room_choice = driver.find_element(By.XPATH, get_xpaths["subpages"]["room_choice"])
+    room_choice.click()
+    WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.XPATH, get_xpaths["subpages"]["room_inc"])))
+
+    for key, value in options.items():
+        driver.find_element(By.XPATH, "//input[@id='" + key + "']").clear()
+        driver.find_element(By.XPATH, "//input[@id='" + key + "']").send_keys(options[key])
```

Comment:
1. Instead of finding the same element twice you can just find it once, store it in a variable and perform both clear() and send_keys()
2. Put that XPATH in a json file. You can use %s for concatenation

Reply:
Changed